crates/bevy_utils/src/lib.rs

```diff
@@ -20,8 +20,12 @@ extern crate alloc;
 /// This includes the most common types in this crate, re-exported for your convenience.
 pub mod prelude {
     pub use crate::default;
+    #[cfg(feature = "alloc")]
+    pub use crate::option_ext::OptionExt as _;
```

Comment:
I think the most controversial aspect of this PR is including the traits in the prelude. We could hold off on this until we're certain that these are used at a frequency to justify it.